Auth.tsx

```diff
@@ -50,10 +53,38 @@ const AuthScreen = () => {
     }
   };
 
+  const handleWhoami = async () => {
+    if (!decryptedData) {
+      console.error("unable to get whoami; must have decrypted data");
+      return;
+    }
+
+    const publicKey = uint8ArrayToHexString(
+      getPublicKey(uint8ArrayFromHexString(decryptedData), true)
+    );
+    const privateKey = decryptedData;
```

Comment:
would be clearer if this was moved before the publicKey computation so that you can write:
```ts
    const publicKey = uint8ArrayToHexString(
      getPublicKey(uint8ArrayFromHexString(privateKey), true)
    );
```
(a tiny bit clearer IMO, because you don't wonder what "decryptedData" is)